specification/storagecache/resource-manager/Microsoft.StorageCache/preview/2023-11-01-preview/amlfilesystem.json

```diff
@@ -0,0 +1,1191 @@
+          },
+          "type": "array"
+        }
+      },
+      "type": "object"
+    },
+    "AmlFilesystemHealth": {
+      "description": "An indication of AML file system health. Gives more information about health than just that related to provisioning.",
+      "readOnly": true,
+      "properties": {
+        "state": {
+          "description": "List of AML file system health states.",
+          "enum": [
+            "Unavailable",
+            "Available",
+            "Degraded",
+            "Transitioning",
+            "Maintenance"
+          ],
+          "x-ms-enum": {
+            "name": "AmlFilesystemHealthStateType",
+            "modelAsString": true
+          },
+          "type": "string"
+        },
+        "statusCode": {
+          "description": "Server-defined error code for the AML file system health",
+          "type": "string"
+        },
+        "statusDescription": {
+          "description": "Describes the health state.",
+          "type": "string"
+        }
+      },
+      "type": "object"
+    },
+    "AmlFilesystemEncryptionSettings": {
+      "description": "AML file system encryption settings.",
+      "properties": {
+        "keyEncryptionKey": {
+          "$ref": "storagecache.json#/definitions/KeyVaultKeyReference",
+          "description": "Specifies the location of the encryption key in Key Vault."
+        }
+      },
+      "type": "object"
+    },
+    "AmlFilesystemHsmSettings": {
+      "description": "AML file system HSM settings.",
+      "x-ms-mutability": [
+        "read",
+        "create"
+      ],
+      "type": "object",
+      "properties": {
+        "container": {
+          "description": "Resource ID of storage container used for hydrating the namespace and archiving from the namespace. The resource provider must have permission to create SAS tokens on the storage account.",
+          "$ref": "storagecache.json#/definitions/URLString"
+        },
+        "loggingContainer": {
+          "description": "Resource ID of storage container used for logging events and errors.  Must be a separate container in the same storage account as the hydration and archive container. The resource provider must have permission to create SAS tokens on the storage account.",
+          "$ref": "storagecache.json#/definitions/URLString"
+        },
+        "importPrefix": {
+          "description": "Only blobs in the non-logging container that start with this path/prefix get hydrated into the cluster namespace.",
+          "type": "string",
+          "default": "/"
+        }
+      },
+      "required": [
+        "container",
+        "loggingContainer"
+      ]
+    },
+    "AmlFilesystemRootSquashSettings": {
+      "description": "AML file system squash settings.",
+      "properties": {
+        "mode": {
+          "type": "string",
+          "description": "Squash mode of the AML file system.",
+          "enum": [
+            "None",
+            "RootOnly",
```

Comment:
add descriptions for enum values

Reply:
Added descriptions for the enum values based on internal team discussion.